enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1031 @@
+    preserving existing headers.
+    * "Replace", which specifies that the IngressController sets the headers,
+    replacing any existing Forwarded or X-Forwarded-* headers.
+    * "IfNone", which specifies that the IngressController sets the headers if
+    they are not already set.
+    * "Never", which specifies that the IngressController never sets the
+    headers, preserving any existing headers. By default, the policy is
+    "Append".
+```
+
+#### Are some headers blocked from being set or deleted?
+  Answer: Please refer to the `Risks and Mitigations` section for more details.
+
+#### Many applications serve mixed content: static assets (images, js, css) and dynamic content (html, json). They usually want to cache the static assets on the client and do not care about session stickiness for those requests. Only the dynamic requests need to stick to the same server (session sticky)
+Most applications will set something like this for static assets:
+```sh
+cache-control: max-age=7200, must-revalidate, public
+```
+But the OpenShift router (HAProxy) will add cache-control: private which results in both headers in the response:
+```sh
+cache-control: private
+cache-control: max-age=7200, must-revalidate, public
+```
+This leads to issues with caching: CDNs like Cloudflare will not cache your files.
+Can we please make sure to only add the `cache-control: private` header if there is no `cache-control: public` header already present in the response?
+  Answer: A common technique is to put static assets on a different virtual host. In OpenShift terms, that means using a distinct route for static assets. This route can
+  be annotated with `haproxy.router.openshift.io/disable_cookies=true` to prevent setting the cookie and thus prevent setting cache-control.
+
+#### Would we be able to [Replace a header by using a regular expression](https://www.haproxy.com/documentation/hapee/latest/traffic-routing/rewrites/rewrite-responses/#replace-a-header-by-using-a-regular-expression) or [Replace part of a header by using a regular expression](https://www.haproxy.com/documentation/hapee/latest/traffic-routing/rewrites/rewrite-responses/#replace-part-of-a-header-by-using-a-regular-expression)?
+  Answer: No, this is out of scope for this enhancement. Action `Set` will perform `set-header` in the haproxy.config for `http-response` and `http-request` which will either set the value if the value is not present or replace the existing value.
+
+#### Should this enhancement allow setting dynamic header values?
+  Answer: Yes, you would be able to set a minimal set of dynamic header values. 
+  We had three options for this EP.
+  1. Allow dynamic header values with all sample fetchers and converters.
+  2. Allow a minimal set of dynamic values that we can confidently validate, focusing on the things that people have asked for on the RFE or [EP](https://github.com/openshift/enhancements/pull/1296#issuecomment-1396103783).
+  3. Prohibit dynamic values entirely.
```

Comment:
```suggestion
#### Should this enhancement allow setting dynamic header values?
  Answer: Yes, you would be able to set a minimal set of dynamic header values. 
  We had three options for this EP.
  
  1. Allow dynamic header values with all sample fetchers and converters.
  2. Allow a minimal set of dynamic values that we can confidently validate, focusing on the things that people have asked for on the RFE or [EP](https://github.com/openshift/enhancements/pull/1296#issuecomment-1396103783).
  3. Prohibit dynamic values entirely.
```

Reply:
do we really need a new line ?

Reply:
anyway done

Reply:
Adding the blank line makes the list show up as an ordered list. Without it, they are all within the previous paragraph.